src/rest_api.py

```diff
@@ -34,15 +34,14 @@ def scan():
 @app.route('/api/v1/register', methods=['POST'])
 def register():
     """
-    Endpoint for registering a new repositor.
+    Endpoint for registering a new repository.
```

Comment:
@abs51295 can you verify that the following logic is implemented for `/register`?

1. Check `OSIORegisteredRepos` table for an entry for the repo name being registered.
2. If there is no entry found:
  a. store the entry in `OSIORegisteredRepos`.
  b. initiate `osioAnalysisFlow` with all required parameters.
  c. return a response in the format specified in the [swagger spec](https://github.com/fabric8-analytics/fabric8-gemini-server/blob/master/swagger.yaml) minus the `last_scanned_at` and `last_scan_report` information as there will not be any available scan report for a newly registered repo.
3. If there is an entry for the repo found in `OSIORegisteredRepos`:
  a. Retrieve the commit hash available in the `OSIORegisteredRepos` table.
  b. Fetch the `worker_results` table with the retrieved commit hash(sha) which is stored as the `external_request_id` and the worker name `ReportGenerationTask`.
  c. If there is no entry found in the `worker_results` table, return a similar response as point no `2c` above.
  d. If there is an entry found in the `worker_results` table, return a result as per the [swagger spec](https://github.com/fabric8-analytics/fabric8-gemini-server/blob/master/swagger.yaml)
  e. For both both the cases `c` and `d`, update the existing entries in the table `OSIORegisteredRepos` with the input paramters `git-sha` and `email-ids`.

Reply:
@samuzzal-choudhury it's ready for review now :)